### PR TITLE
event: add location

### PR DESCRIPTION
**Pull request by Joe2k**
For better visibility of tournaments. Also I'm gonna keep the events table as the base to build the tournament stuff on. So location will be very useful then

### CODE REVIEW

File: server/management/commands/import_uc_events.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ def handle(self, *args: Any, **options: Any) -> None:
                     "start_date": tournament["start"],
                     "end_date": tournament["end"],
                     "ultimate_central_slug": tournament["slug"],
+                    "location": tournament["Location"]["formatted_address"],
```
Comment (punchagan) [inline server/management/commands/import_uc_events.py:41]: Why the formatted address? Is there no better location info?

File: server/migrations/0026_merge_0025_event_location_0025_player_imported_data.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Generated by Django 4.2.2 on 2023-08-30 12:52
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("server", "0025_event_location"),
```
Comment (punchagan) [inline server/migrations/0026_merge_0025_event_location_0025_player_imported_data.py:8]: Working with merge migrations could be a mess. Can you rollback the migrations to `0024`, and recreate them to create a linear migration dependency?

**Review by punchagan — CHANGES_REQUESTED** [review]
Yes, I copied over only basic information from UC so that we can add more stuff, as and when we need it. Left a couple of comments.

File: server/management/commands/import_uc_events.py (hunk 1 above)
Comment (Joe2k) [reply server/management/commands/import_uc_events.py:41]:
```
"Location": {
                "model": "location",
                "id": 732849,
                "address_1": null,
                "address_2": null,
                "locality": "Bangalore",
                "region": "Karnataka",
                "postal_code": null,
                "country": "IN",
                "latitude": null,
                "longitude": null,
                "timezone": null,
                "formatted_address": "Bangalore, Karnataka, IN",
                "is_manual_entered": true,
                "created_at": "2021-01-01 07:24:17",
                "updated_at": "2021-01-01 07:24:17"
            },
```

This is the example data we get. In this the most complete one is formatted_address for most

File: server/migrations/0026_merge_0025_event_location_0025_player_imported_data.py (hunk 2 above)
Comment (Joe2k) [reply server/migrations/0026_merge_0025_event_location_0025_player_imported_data.py]: did this and pushed now

File: server/management/commands/import_uc_events.py (hunk 1 above)
Comment (punchagan) [reply server/management/commands/import_uc_events.py:41]: Do you think they'll ever start adding lat/lng? We can do funky map things if we have that info... but I guess for most things we won't?

Comment (punchagan) [reply server/management/commands/import_uc_events.py:41]: I guess we can add lat/lng when we need it, from UC data or doing some parsing on our end, maybe. :shrug:

Comment (Joe2k) [reply server/management/commands/import_uc_events.py:41]: So you are suggesting to add that column now itself and use when needed right?

Comment (Joe2k) [reply server/management/commands/import_uc_events.py:41]: But yes agree we can do some fun stuff with that. But not sure if well do it immediatly now

Comment (punchagan) [reply server/management/commands/import_uc_events.py]: Let's do it later.

**Comment by punchagan** [comment]
Merged after adding a preparatory commit to fix how we create/update events in the DB, and splitting your commits into atomic ones, @Joe2k !  Thanks! :tada:

**Pull request merged**